tests/console/firewalld.pm

```diff
@@ -27,59 +27,82 @@ sub pre_test {
-
-    # Default is for Tumbleweed and newer SLE/Leap
-    if (!is_sle('<15-SP3') && !is_leap('<15.3')) {
-        assert_script_run("firewall-cmd --zone=public --add-port=25/tcp");
-        assert_script_run("iptables -C IN_public_allow -p tcp --dport 25 -m conntrack --ctstate NEW,UNTRACKED -j ACCEPT");
-
-        assert_script_run("firewall-cmd --zone=public --add-service=pop3");
-        assert_script_run("iptables -C IN_public_allow -p tcp --dport 110 -m conntrack --ctstate NEW,UNTRACKED -j ACCEPT");
-
-        assert_script_run("firewall-cmd --zone=public --add-protocol=icmp");
-        assert_script_run("iptables -C IN_public_allow -p icmp -m conntrack --ctstate NEW,UNTRACKED -j ACCEPT");
-
-        assert_script_run("firewall-cmd --zone=public --add-port=2000-3000/udp");
-        assert_script_run("iptables -C IN_public_allow -p udp --dport 2000:3000 -m conntrack --ctstate NEW,UNTRACKED -j ACCEPT");
+    if (!is_tumbleweed) {
+        assert_script_run("iptables -L IN_public_allow --line-numbers | sed '/^num\\|^\$\\|^Chain/d' | wc -l > /tmp/nr_rules.txt");
     }
     else {
-        assert_script_run("firewall-cmd --zone=public --add-port=25/tcp");
-        assert_script_run("iptables -C IN_public_allow -p tcp --dport 25 -m conntrack --ctstate NEW -j ACCEPT");
+        assert_script_run("nft list chain inet firewalld filter_IN_public_allow | wc -l > /tmp/nr_in_public.txt");
+        assert_script_run("nft list chain inet firewalld filter_FWDI_public | wc -l > /tmp/nr_fwdi_public.txt");
+    }
 
-        assert_script_run("firewall-cmd --zone=public --add-service=pop3");
-        assert_script_run("iptables -C IN_public_allow -p tcp --dport 110 -m conntrack --ctstate NEW -j ACCEPT");
+    assert_script_run("firewall-cmd --zone=public --add-port=25/tcp");
+    assert_script_run("firewall-cmd --zone=public --add-service=pop3");
+    assert_script_run("firewall-cmd --zone=public --add-protocol=icmp");
+    assert_script_run("firewall-cmd --zone=public --add-port=2000-3000/udp");
 
-        assert_script_run("firewall-cmd --zone=public --add-protocol=icmp");
-        assert_script_run("iptables -C IN_public_allow -p icmp -m conntrack --ctstate NEW -j ACCEPT");
-
-        assert_script_run("firewall-cmd --zone=public --add-port=2000-3000/udp");
-        assert_script_run("iptables -C IN_public_allow -p udp --dport 2000:3000 -m conntrack --ctstate NEW -j ACCEPT");
-    }
+    check_rules;
 
     # Reload default configuration
     record_info 'Reload default configuration';
     assert_script_run("firewall-cmd --reload");
-    assert_script_run("if [ `iptables -L IN_public_allow --line-numbers | sed '/^num\\|^\$\\|^Chain/d' | wc -l` -eq `cat /tmp/nr_rules.txt` ]; then /usr/bin/true; else /usr/bin/false; fi");
+    if (!is_tumbleweed) {
+        assert_script_run("if [ `iptables -L IN_public_allow --line-numbers | sed '/^num\\|^\$\\|^Chain/d' | wc -l` -eq `cat /tmp/nr_rules.txt` ]; then /usr/bin/true; else /usr/bin/false; fi");
```

Comment:
```suggestion
        assert_script_run("[ `iptables -L IN_public_allow --line-numbers | sed '/^num\\|^\$\\|^Chain/d' | wc -l` -eq `cat /tmp/nr_rules.txt` ]");
```

Reply:
Good idea - I will handle that in another PR

Reply:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12277